packages/web-components/src/components/image-with-caption/image-with-caption.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html, property, customElement, LitElement } from 'lit-element';
+import settings from 'carbon-components/es/globals/js/settings';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings.js';
+import '../image/image';
+import '../lightbox-media-viewer/lightbox-image-viewer';
+import '../button/button';
+import ZoomIn20 from 'carbon-web-components/es/icons/zoom--in/20';
+import DDSModal from '../modal/modal';
+import 'carbon-web-components/es/components/modal/modal-close-button';
+import styles from './image-with-caption.scss';
+
+const { prefix } = settings;
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * Image With Caption
+ *
+ * @element dds-image-with-caption
+ */
+
+@customElement(`${ddsPrefix}-image-with-caption`)
+class DDSImageWithCaption extends LitElement {
+  /**
+   * The lightbox.
+   */
+  @property({ type: Boolean, reflect: true })
+  lightbox = false;
+
+  /**
+   * The image source.
+   */
+  @property({ reflect: true })
```

Comment:
We use dash case (instead of camel case in properties) for attribute names:

```suggestion
  @property({ reflect: true, attribute: 'default-src' })
```